pkg/internal/request/span.go

```diff
@@ -25,22 +25,22 @@ var clocks = converter{monoClock: monotime.Now, clock: time.Now}
 // Span contains the information being submitted by the following nodes in the graph.
 // It enables comfortable handling of data from Go.
 type Span struct {
-	Type          EventType
-	ID            uint64
-	Method        string
-	Path          string
-	Route         string
-	Peer          string
-	Host          string
-	HostPort      int
-	Status        int
-	ContentLength int64
-	RequestStart  int64
-	Start         int64
-	End           int64
-	ServiceName   string
-	Metadata      []MetadataTag
-	TraceID       string
+	Type           EventType
+	ID             uint64
+	Method         string
+	Path           string
+	Route          string
+	Peer           string
+	Host           string
+	HostPort       int
+	Status         int
+	ContentLength  int64
+	RequestStart   int64
+	Start          int64
+	End            int64
+	ServiceName    string
+	Metadata       []MetadataTag
+	TraceparentPtr *[55]byte
```

Comment:
+1. I'd use a string, which at the end is a pointer to a byte slice, with some extra comfort

Reply:
This change was prompted by: 

> Do you think it's better if we moved this logic now completely to the traces code? We do this extra allocation here of technically almost the same size as the incoming array, and it's all wasted if we are only producing metrics. We can simply let the original byte array get all the way to the traces code and have the array slicing done there.

Maybe I took that too literally?  I don't like that it's an odd untyped (unabstracted) array w/ magic number and would like to change that.  I am probably thinking too much like a C programmer:  If we make request.Span have a string instead of this pointer, and then do something like: "Traceparent: string(trace.Traceparent)" does that just allocate the underlying string's slice (length/pointer) without doing a copy of the data?  In other words, what is the memory size and performance difference between saving a pointer to byte slice, vs. saving a string?  Is it just that the string has a length to it?  If so, then I'll definitely change this to a string.  Thanks!

Reply:
Yeah small overhead to add a string and when the parent is empty, many many situations, we'll get nothing on the struct.

Reply:
Converted to string.